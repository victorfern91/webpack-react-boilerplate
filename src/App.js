import React from "react";
import ReactDOM from "react-dom";

let hot = null;

const App = () => (
    <h1>Hello World!</h1>
);

if (process.env.NODE_ENV === "development") {
     hot = require("react-hot-loader").hot;
}

export default process.env.NODE_ENV === "development"
    ? hot(module)(App)
    : App;
