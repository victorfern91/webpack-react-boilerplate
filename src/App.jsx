import React from 'react';
import { HelloLazyWorld } from 'components/LazyComponents';
import { includeHotReloadCapabilities } from 'utils/dev';
import logo from 'static/logo.svg';

const App = () => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>
        Edit <code>src/App.js</code> and save to reload.
      </p>
      <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
        Learn React
      </a>
      <HelloLazyWorld />
    </header>
  </div>
);

export default includeHotReloadCapabilities(App);
