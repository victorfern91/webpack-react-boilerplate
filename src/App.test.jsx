import React from 'react';
import { shallow } from 'enzyme';
import App from 'App';

jest.mock('utils/dev');

describe('App', () => {
  it('should render something', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
