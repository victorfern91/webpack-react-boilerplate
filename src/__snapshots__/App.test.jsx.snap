// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render something 1`] = `
<h1>
  Hello World!
</h1>
`;
